sphinx/domains/math.py

```diff
@@ -93,8 +93,9 @@ def resolve_xref(self, env: BuildEnvironment, fromdocname: str, builder: Builder
                      typ: str, target: str, node: pending_xref, contnode: Element
                      ) -> Element | None:
         assert typ in ('eq', 'numref')
-        docname, number = self.equations.get(target, (None, None))
-        if docname:
+        result = self.equations.get(target)
+        if result:
+            docname, number = result
```

Comment:
can we keep the tuple unpacking here?

Reply:
no, the tuple unpacking has to be inside the `if` statement for this to be correctly typed.

I could potentially use the walrus operator here, which would look something like-

```python
if (result := self.equations.get(target)):
    docname, number = result
```